chain/app/app.go

```diff
@@ -296,14 +298,33 @@ func NewBandApp(
 	// The initChainer handles translating the genesis.json file into initial state for the network
 	app.SetInitChainer(app.InitChainer)
 	app.SetBeginBlocker(app.BeginBlocker)
+
+	// Create default an AnteHandler that will be used after checking refundGasPrice
+	defaultAnteHandler := auth.NewAnteHandler(
+		app.accountKeeper,
+		app.supplyKeeper,
+		auth.DefaultSigVerificationGasConsumer,
+	)
 	// The AnteHandler handles signature verification and transaction pre-processing
 	app.SetAnteHandler(
-		auth.NewAnteHandler(
-			app.accountKeeper,
-			app.supplyKeeper,
-			auth.DefaultSigVerificationGasConsumer,
-		),
+		func(ctx sdk.Context, tx sdk.Tx, simulate bool) (newCtx sdk.Context, res sdk.Result, abort bool) {
+			if stdTx, ok := tx.(auth.StdTx); ok {
+				fee := stdTx.Fee
+				for _, msg := range tx.GetMsgs() {
+					if report, ok := msg.(zoracle.MsgReportData); ok {
+						refundGasPrice := sdk.NewCoins(sdk.NewCoin("uband", sdk.NewInt(report.RefundGasPrice)))
+						_, refundGasPriceExceedTxGasPrice := fee.GasPrices().SafeSub(sdk.NewDecCoins(refundGasPrice))
```

Comment:
Can we use this function instead of Safesub? https://github.com/cosmos/cosmos-sdk/blob/master/types/coin.go#L380

Reply:
Sadly we are comparing fee.GasPrices() and refundGasPrice which are DecCoins and have no such a function.